### PR TITLE
Client chain elements

### PR DESCRIPTION
**Pull request by Bolodya1997**
1. Add 'interfaceName' to kernel mechanism client
2. ~Add memif mechanism client~

### CODE REVIEW

File: pkg/networkservice/common/mechanisms/memif/client.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
```
Comment (edwarnicke) [inline pkg/networkservice/common/mechanisms/memif/client.go:1]: I'm a bit confused why this is here... what is it attaching to the memif?  You need something (vpp or something else) to attach to memif...

Comment (Bolodya1997) [reply pkg/networkservice/common/mechanisms/memif/client.go]: OK, guess it is not the right place/time for the memif client :)

File: pkg/networkservice/common/mechanisms/kernel/client.go [hunk 2]
```diff
@@ -21,19 +21,24 @@ import (
...
+type kernelMechanismClient struct {
+	interfaceName string
+}
 
 // NewClient - returns client that sets kernel preferred mechanism
-func NewClient() networkservice.NetworkServiceClient {
-	return &kernelMechanismClient{}
+func NewClient(interfaceName string) networkservice.NetworkServiceClient {
+	return &kernelMechanismClient{
+		interfaceName: interfaceName,
```
Comment (denis-tingajkin) [inline pkg/networkservice/common/mechanisms/kernel/client.go:40]: As I can see we can work with `interfaceName== ""` https://github.com/networkservicemesh/api/blob/e5aee3ba99efe11afe73263b9d2e4396da3da1f7/pkg/api/networkservice/mechanisms/kernel/helpers.go#L92
So I would recommend adding an option for this chain element to set some custom interface name.

**Review by denis-tingajkin — CHANGES_REQUESTED** [review]

File: pkg/networkservice/common/mechanisms/kernel/client.go (hunk 2 above)
Comment (Bolodya1997) [reply pkg/networkservice/common/mechanisms/kernel/client.go:40]: It is actually an option, user can set `interfaceName== ""` if there is no need for some custom interface name.

Comment (denis-tingajkin) [reply pkg/networkservice/common/mechanisms/kernel/client.go:40]: I think better if the user do not need to pass anything into the constructor.

Comment (Bolodya1997) [reply pkg/networkservice/common/mechanisms/kernel/client.go]: Yep, done

File: pkg/networkservice/common/mechanisms/kernel/option.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+// limitations under the License.
+
+package kernel
+
+type option struct {
+	interfaceName string
+}
+
+// Option for kernel mechanism
+type Option func(*option)
```
Comment (denis-tingajkin) [inline pkg/networkservice/common/mechanisms/kernel/option.go:24]:
```suggestion
type Option func(*kernelMechanismClient )
```

**Review by denis-tingajkin — APPROVED** [review]

**Pull request merged**